network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/TerminalImpl.java

```diff
@@ -90,7 +91,10 @@ public Terminal setQ(double q) {
 
     @Override
     public double getI() {
-        throw new UnsupportedOperationException("TODO");
+        if (connectable.getType() == ConnectableType.BUSBAR_SECTION) {
+            return 0;
+        }
+        return isConnected() ? Math.hypot(getP(), getQ()) / (Math.sqrt(3.) * getBusView().getBus().getV() / 1000) : 0;
```

Comment:
Why do we check for connection?

Reply:
getBusView().getBus() is null if terminal is not connected